### ISSUE TITLE
FR: Make StreamView support the a const constructor.

### ISSUE THREAD

**Issue by matanlurey** [dart-lang/sdk#25760]
`StreamView` acts as a delegating Stream implementation and has no internal state.

Can we make it `const`?

Assume you have the following class:

```dart
abstract class CustomStream implements Stream { ... }
```

It would be nice to re-use `Stream.empty` when creating a no-op version:

```dart
abstract class NoopCustomStream extends StreamView implements CustomStream {
  const NoopCustomStream() : super(const Stream.empty());
}
```

This is not possible today because `StreamView` is not a `const` constructor.

**Comment by floitschG** [comment]
I think that's reasonable.
I created a CL: https://codereview.chromium.org/1696453002/

Lasse is currently not able to review (and I would like his opinion on it), so it might take some time before I will commit.
Please ping in ~3 weeks if nothing happened.

**Comment by floitschG** [comment]
Fixed with 257272fd42a5ec7e48026cc0157b5be0a1308cab

**Comment by matanlurey** [comment]
I really appreciate it, thanks!
